src/frontend/src/auth.ts

```diff
@@ -88,7 +88,9 @@ async function handleAuthRequest(
   request: AuthRequest,
   hostname: FrontendHostname
 ): Promise<AuthResponse> {
-  if (!(await confirmRedirect(hostname))) {
+  const userPrincipal = await connection.getPrincipal(userNumber, hostname);
```

Comment:
Do we need to cover this with a spinner? Query calls are fast, but depending on your connection you might still end up with an "unresponsive" app.

Reply:
Are you thinking the "astronaut" kind of spinner or a new kind?

Generally this makes sense though!

Reply:
Yeah the astronaut. I do the same when loading the data for the `manage` page with a query

Reply:
Done! PTAL.